QuizApp/Quiz/Leaderboard/LeaderboardViewController.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  LeaderboardViewController.swift
+//  QuizApp
+//
+//  Created by five on 14.05.2021..
+//
+
+import UIKit
+
+class LeaderboardViewController: UIViewController {
+    
+    
+    let cellId = "leaderboardCellId"
+    private var router: AppRouter!
+    private var quiz: Quiz!
+    private var leaderboardData: [LeaderboardResult]!
+    
+    convenience init(router: AppRouter, quiz: Quiz) {
+        self.init()
+        self.router = router
+        self.quiz = quiz
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupView()
+        NetworkService().fetchLeaderboard(quiz: quiz, completionHandler: { [self] (result: Result<[LeaderboardResult], RequestError>) -> Void in
+            switch result {
+            case .success(let leaderboardResults):
+                leaderboardData = leaderboardResults
+                DispatchQueue.main.async {
+                    tableView.reloadData()
+                }
+            case .failure(let error):
+                handleError(error)
+            }
+        })
```

Comment:
Even though it is only executed once, it would be better to extract it to a method so that viewDidLoad would be more readable.